Comment for testing build

TomerAdmon opened the pull request:
now we can trigger build with those properties:
- personal build
- comment
- queue on top
- clean sources
- require rebuild dependencies

acceptance tests were fixed.

borismod commented on FluentTc/Locators/MoreOptionsHavingBuilder.cs:
```diff
@@ -0,0 +1,77 @@
+        IMoreOptionsHavingBuilder WithComment(string comment);
+    }
+
+    public class MoreOptionsHavingBuilder : IMoreOptionsHavingBuilder
```
MoreOptionsHavingBuilder should not be visible by user of FluentTc. Usually such classes are internal

borismod commented on FluentTc/Engine/BuildConfigurationRunner.cs:
```diff
@@ -49,17 +63,48 @@ private static List<Property> GetProperties(Action<IBuildParameterValueBuilder>
+            if (moreOptions != null &&
+                moreOptions.GetComment() != null)
+            {
+                bodyBuilder.AppendFormat(@"<comment><text>{0}</text></comment>", moreOptions.GetComment()).AppendLine();
```
What happens if comment contains some characters that might break the XML, like < > /?

borismod commented on FluentTc/Engine/BuildConfigurationRunner.cs:
```diff
@@ -49,17 +63,48 @@ private static List<Property> GetProperties(Action<IBuildParameterValueBuilder>
+            }
+
+            if (moreOptions != null &&
+                (moreOptions.TriggeringOptions.CleanSources == true || 
```
There is no need to check booleans with "==true", it's enough to check:
```C# 
if ( moreOptions.TriggeringOptions.CleanSources )
{
}
```

borismod replied:
Don;t you need also to check 
```C#
if ( moreOptions.TriggeringOptions.HasValue )
```

borismod commented:
@TomerAdmon, great contribution :+1: 
I wrote a few minor notes, could you take a look :question:

TomerAdmon commented:
Hi, 
Thanks for the review.
1) fixed
2) fixed + add a test
3) regarding the ==true, since this is a nullable bool, we need to use the ==True (see [here](http://stackoverflow.com/questions/2673918/best-way-to-check-for-nullable-bool-in-a-condition-expression-if))
4) regarding the "if ( moreOptions.TriggeringOptions.HasValue )", we check that the moreOptions is not null, and if it's not null it means we initialized the ctor that created a new TriggeringOptions... so no need to test it....

borismod commented on FluentTc/Locators/MoreOptionsHavingBuilder.cs:
```diff
@@ -0,0 +1,77 @@
+        }
+    }
+
+    public class TriggeringOptions
```
Can be also internal, isn't it?
